StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethods/SavedPaymentMethodsSheet.swift

```diff
@@ -0,0 +1,233 @@
+            assertionFailure("presentingViewController is already presenting a view controller")
+            let error = SavedPaymentMethodsSheetError.unknown(
+                debugDescription: "presentingViewController is already presenting a view controller"
+            )
+            savedPaymentMethodsSheetDelegate?.didFail(with: error)
+            return
+        }
+        loadPaymentMethods { result in
+            switch result {
+            case .success(let savedPaymentMethods):
+                self.present(from: presentingViewController, savedPaymentMethods: savedPaymentMethods)
+            case .failure(let error):
+                self.savedPaymentMethodsSheetDelegate?.didFail(with: .errorFetchingSavedPaymentMethods(error))
+                return
+            }
+        }
+        presentingViewController.presentAsBottomSheet(bottomSheetViewController,
+                                                      appearance: configuration.appearance)
+    }
+
+    @available(iOSApplicationExtension, unavailable)
+    @available(macCatalystApplicationExtension, unavailable)
+    func present(from presentingViewController: UIViewController,
+                 savedPaymentMethods: [STPPaymentMethod]) {
+        let loadSpecsPromise = Promise<Void>()
+        AddressSpecProvider.shared.loadAddressSpecs {
+            loadSpecsPromise.resolve(with: ())
+        }
+
+        loadSpecsPromise.observe { _ in
+            DispatchQueue.main.async {
+                let isApplePayEnabled = StripeAPI.deviceSupportsApplePay() && self.configuration.applePayEnabled
+                let savedPaymentSheetVC = SavedPaymentMethodsViewController(savedPaymentMethods: savedPaymentMethods,
+                                                                            configuration: self.configuration,
+                                                                            isApplePayEnabled: isApplePayEnabled,
+                                                                            savedPaymentMethodsSheetDelegate: self.savedPaymentMethodsSheetDelegate,
+                                                                            delegate: self)
+                self.bottomSheetViewController.contentStack = [savedPaymentSheetVC]
+            }
+        }
+    }
+    // MARK: - Internal Properties
+    var completion: (() -> Void)?
+}
+
+extension SavedPaymentMethodsSheet {
+    func loadPaymentMethods(completion: @escaping (Result<[STPPaymentMethod], SavedPaymentMethodsSheetError>) -> Void) {
+        configuration.customerContext.listPaymentMethodsForCustomer {
+            paymentMethods, error in
+            guard let paymentMethods = paymentMethods, error == nil else {
+                // TODO: Pass errors from the customerContext
+                let error = PaymentSheetError.unknown(debugDescription: "Failed to retrieve PaymentMethods for the customer")
+//                let error = error ?? PaymentSheetError.unknown(
```

Comment:
Delete?

Reply:
resolving-- todo is still there, but i have a comment on it below.